src/bun.js/node/path_watcher.zig

```diff
@@ -111,6 +116,13 @@ pub const PathWatcherManager = struct {
             .vm = vm,
             .watcher_count = 0,
             .mutex = Mutex.init(),
+            // we schedule directory search to run on a thread pool because we dont want to block the main thread
+            // threads are only spawned when some task is schedule
+            .thread_pool = ThreadPool.init(.{
```

Comment:
don't use a dedicated thread pool for this

just use JSC.WorkPool

Reply:
Fixed on https://github.com/oven-sh/bun/pull/3846/commits/b3995906c528ae6b40cb211c59ed29bec5e2b5ba